setup.py

```diff
@@ -0,0 +1,27 @@
+# This file is part of ts_fbs_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import setuptools_scm
+from setuptools import setup
+
+setup(
+    version=setuptools_scm.get_version(),
```

Comment:
Confused. Shouldn't this go in the pyproject.toml? Is there something about the conda build system that requires you to use setup.py?

Reply:
This allows the conda recipe to automatically generate the version string from `git describe`. Unfortunately, it is not supported by `pyproject.toml` yet, so we need to rely on `setup.py` for now.

Reply:
setuptools_scm normally works fine from pyproject.toml. I was wondering if this was conda specific.

```toml
[build-system]
requires = [
   "setuptools",
   "setuptools_scm",
]
```

works fine normally, and a `setuptools_scm` section in the pypeoject.toml as well and something like:

```python
__all__ = ["__version__"]

from importlib.metadata import PackageNotFoundError, version

__version__: str
"""The application version string (PEP 440 / SemVer compatible)."""

try:
    __version__ = version("example")
except PackageNotFoundError:
    # package is not installed
    __version__ = "0.0.0"
```
in the `__init__.py` somewhere.

Reply:
To my best of knowledge this doesn't work with conda, other than the way Tiago implemented it. See also https://github.com/conda/conda-build/issues/4774